### PR TITLE
Use nullable values rather than js.UndefOr's in java.net.URI.

### PR DESCRIPTION
**Pull request by sjrd**
Extracted from #4666.

---

This is a step towards removing more dependencies from the javalib on the Scala library.

This actually makes the code somewhat simpler in a number of places, because the public API of that class uses nullable values anyway. Although it does make the regex extraction a bit more complex.

### CODE REVIEW

File: javalib/src/main/scala/java/net/URI.scala [hunk 1]
```diff
@@ -35,38 +35,60 @@ final class URI(origStr: String) extends Serializable with Comparable[URI] {
   if (_fld == null)
     throw new URISyntaxException(origStr, "Malformed URI")
 
-  private val _isAbsolute = fld(AbsScheme).isDefined
-  private val _isOpaque = fld(AbsOpaquePart).isDefined
+  private val _isAbsolute = _fld(AbsScheme).isDefined
+  private val _isOpaque = _fld(AbsOpaquePart).isDefined
 
-  @inline private def fld(idx: Int): js.UndefOr[String] = _fld(idx)
+  @inline private def fld(idx: Int): String = _fld(idx).orNull
```
Comment (gzm0) [inline javalib/src/main/scala/java/net/URI.scala:41]: Also remove this usage of `js.UndefOr`?

**Review by gzm0 — APPROVED** [review]
Only one nit.

File: javalib/src/main/scala/java/net/URI.scala (hunk 1 above)
Comment (sjrd) [reply javalib/src/main/scala/java/net/URI.scala:41]: Oh, I had forgotten about this PR :s Sorry about that.

There's not only this spot but also the two calls to `_fld(...).isDefined` above. I can't get rid of them without obscuring the code, at the moment. The intent is to replace them by calls to helpers similar to
https://github.com/scala-js/scala-js/blob/d02805412faeeef620e203b9e98ca966067db648/javalanglib/src/main/scala/java/lang/Utils.scala#L21-L23
in a later PR, where we make a full swipe of all remaining `js.UndefOr` operations all over the javalib.

Alternatively, I could start a `Utils.scala` helper class in this PR with only `isDefined` and `orNull`, and expand it later with more stuff.

WDYT?

Comment (gzm0) [reply javalib/src/main/scala/java/net/URI.scala:41]: Yeah, if that is the plan, doing it later is fine.

**Pull request merged**